ext/soap/soap.c

```diff
@@ -1146,8 +1145,8 @@ PHP_METHOD(SoapServer, addFunction)
 		key = zend_string_tolower(Z_STR_P(function_name));
 
 		if ((f = zend_hash_find_ptr(EG(function_table), key)) == NULL) {
-			php_error_docref(NULL, E_WARNING, "Tried to add a non existent function '%s'", Z_STRVAL_P(function_name));
-			return;
+			zend_argument_type_error(1, "must be a valid callback, function \"%s\" not found", Z_STRVAL_P(function_name));
```

Comment:
This error message is a bit misleading, in that it implies that it accepts arbitrary callbacks (not just global functions).

Reply:
Changed it to "must be a valid function name", if that's ok for you